kube/src/client/mod.rs

```diff
@@ -328,17 +327,19 @@ fn handle_api_errors(text: &str, s: StatusCode) -> Result<()> {
     if s.is_client_error() || s.is_server_error() {
         // Print better debug when things do fail
         // trace!("Parsing error: {}", text);
-        if let Ok(errdata) = serde_json::from_str::<ErrorResponse>(text) {
+        if let Ok(errdata) = serde_json::from_str::<Status>(text) {
             debug!("Unsuccessful: {:?}", errdata);
             Err(Error::Api(errdata))
         } else {
             warn!("Unsuccessful data error parse: {}", text);
             // Propagate errors properly via reqwest
-            let ae = ErrorResponse {
-                status: s.to_string(),
-                code: s.as_u16(),
-                message: format!("{:?}", text),
-                reason: "Failed to parse error data".into(),
+            let ae = Status {
+                metadata: Default::default(),
+                status: Some(s.to_string()),
+                code: Some(s.as_u16().into()),
+                message: Some(format!("{:?}", text)),
+                reason: Some("Failed to parse error data".into()),
+                details: None,
```

Comment:
This now causes a lot more option types that always exist (like most k8s-openapi structs), and that was the main reason for the separation of the structs.